app/pagination.py

```diff
@@ -38,19 +41,26 @@ async def paginate[M: Base, S: BaseSchema](
     schema_cls: type[S],
     *,
     extra_conditions: list[ColumnExpressionArgument] | None = None,
+    options: list[ORMOption] | None = None,
 ) -> AbstractPage[S]:
+    """
+    This function queries a database model (M) using a ModelHandler.
+    It applies optional filtering (extra_conditions) and query options (options).
+    It then serializes the results into a schema (S) and returns
+    a paginated response in the form of AbstractPage[S].
+    """
     params: LimitOffsetParams = resolve_params()
-    extra_conditions = extra_conditions or []
-
-    total = await handler.count(*extra_conditions)
+    total = await handler.count(*(extra_conditions or []))
     items: Sequence[M] = []
     if params.limit > 0:
         items = await handler.fetch_page(
-            limit=params.limit, offset=params.offset, extra_conditions=extra_conditions
+            limit=params.limit,
+            offset=params.offset,
+            extra_conditions=extra_conditions,
+            page_options=options,
```

Comment:
in paginate the name of the function argument should be page_options, in the handler either options either orm_options. Is it what you requested @arturbalabanov  ?